sdk/monitor/Azure.Monitor.Query/src/Models/MonitorQueryModelFactory.cs

```diff
@@ -36,5 +36,35 @@ public static LogsQueryResult LogsQueryResult(IReadOnlyList<LogsTable> allTables
             JsonElement errorJson = error.ToObjectFromJson<JsonElement>();
             return new LogsQueryResult(allTables, statisticsJson, visualizationJson, errorJson);
         }
+
+        /// <summary> Initializes a new instance of LogsTableRow. </summary>
+        /// <returns> A new <see cref="Models.LogsTableColumn"/> instance for mocking. </returns>
+        public static LogsTableRow LogsTableRow(IReadOnlyList<LogsTableColumn> columns, Object[] rows)
+        {
+            Dictionary<string, int> columnDictionary = new();
+
+            for (var index = 0; index < columns.Count; index++)
+            {
+                columnDictionary[columns[index].Name] = index;
+            }
+
+            JsonElement row = JsonElementFromObject(rows);
+            return new LogsTableRow(columnDictionary, columns, row);
+        }
+
+        /// <summary> Initializes a new instance of LogsTable. </summary>
+        /// <exception cref="ArgumentNullException"> <paramref name="name"/> or <paramref name="rows"/> is null. </exception>
+        public static LogsTable LogsTable(string name, IEnumerable<LogsTableRow> rows, IEnumerable<LogsTableColumn> columns)
+        {
+            JsonElement row = JsonElementFromObject(rows);
+            return new LogsTable(name, columns, row);
+        }
+
+        private static JsonElement JsonElementFromObject<TValue>(TValue value, JsonSerializerOptions options = default)
```

Comment:
Nit - this might be worth making into a shared source extension method on `BinaryData` like `public static JsonElement AsJsonElement(this BinaryData data)` so it could be easily reused elsewhere in our libraries via `BinaryData.FromObjectAsJson<T>(value, options).AsJsonElement()`.  Don't do it as part of this PR, but keep your eyes peeled for anyone else doing this too and maybe we can share an abstraction.